docs/proposals/disk-accounting.md

```diff
@@ -0,0 +1,641 @@
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/proposals/disk-accounting.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+**Author**: Vishnu Kannan
+
+**Last** **Updated**: 11/5/2015
+
+**Status**: Pending Review
+
+This proposal is an attempt to come up with a means for accounting disk usage in Kubernetes clusters that are running docker as the container runtime. Some of the principles here might apply for other runtimes too.
+
+### Why is disk accounting necessary?
+
+Present kubernetes clusters become unusable over time due to the local disk becoming full. The kubelets on the node attempt to perform garbage collection of old containers and images, but that doesn’t prevent running pods from using up all the available disk space.
+
+Kubernetes users have no insight into how the disk is being consumed.
+
+Large images and rapid logging, can lead to temporary downtime on the nodes. The node has to free up disk space by deleting images and containers. During this cleanup, existing pods can fail and new pods cannot be started. The node will also transition into an `OutOfDisk` condition, preventing more pods from being scheduled to the node.
+
+Automated eviction of pods that are hogging the local disk is not possible since proper accounting isn’t available.
+
+Since local disk is a non-compressible resource, users need means to restrict usage of local disk by pods and containers. Proper disk accounting is a prerequisite. As of today, a misconfigured low QoS class pod, can end up bringing down the entire cluster by taking up all the available disk space (misconfigured logging for example)
+
+### Goals
+
+1. Account for disk usage on the nodes.
+
+2. Compatibility with the most important docker storage backends - devicemapper, aufs and overlayfs
+
+3. Provide a roadmap for enabling disk as a schedulable resource in the future.
+
+4. Provide a plugin interface for extending support to non-default filesystems and storage drivers.
+
+### Non Goals
+
+1. Compatibility with all storage backends. The matrix is pretty large already and the priority is to get disk accounting to on most widely deployed platforms.
+
+2. Support for filesystems other than ext4.
```

Comment:
Thanks for the update. This change will affect the writable layer and the image layers. Since we depend on devicemapper semantics here, this change shouldn't affect this proposal.
I assume that the primary filesystem where volumes and logs will be stored will continue to be ext4? This is important because, the current plan is to use quota for those use cases. Quota configuration is slightly different between ext4 and xfs.